app/services/integrations/aggregator/taxes/avalara/fetch_company_id_service.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Integrations
+  module Aggregator
+    module Taxes
+      module Avalara
+        class FetchCompanyIdService < Integrations::Aggregator::BaseService
+          def action_path
+            "v1/#{provider}/companies"
+          end
+
+          def call
+            throttle!(:avalara)
+
+            response = http_client.post_with_response(payload, headers)
+            body = JSON.parse(response.body)
+
+            received_company = body["companies"]&.first
+
+            if received_company.blank?
+              code = "company_not_found"
+              message = "Company cannot be found in Avalara based on the provided code"
+
+              deliver_integration_error_webhook(integration:, code:, message:)
+
+              return result.service_failure!(code:, message:)
+            end
+
+            result.company = OpenStruct.new(received_company)
```

Comment:
We are trying to get rid of `OpenStruct` from the project, any chance it can be replaces by an other type of data structure? (Data, Struc, Hash...)

Reply:
Using Hash now since we get it directly from `JSON.parse`